modules/expo-bluesky-video-player/ios/ExpoBlueskyVideoPlayerView.swift

```diff
@@ -0,0 +1,65 @@
+import ExpoModulesCore
+
+public class ExpoBlueskyVideoPlayerView: ExpoView, AVPlayerViewControllerDelegate {
+  let onPlayerStateChange = EventDispatcher()
+  
+  private var controller: PlayerController
+  
+  var source: String? = nil
+  var autoplay = true
+  var isPlaying: Bool
+  
+  public override var bounds: CGRect {
+    didSet {
+      controller.setFrame(rect: bounds)
+    }
+  }
+  
+  public required init(appContext: AppContext? = nil) {
+    self.isPlaying = self.autoplay
+    self.controller = PlayerController()
+    super.init(appContext: appContext)
+    self.clipsToBounds = true
+    
+    self.controller.setFrame(rect: bounds)
+    self.addSubview(self.controller.view)
+  }
+  
+  public override func willMove(toWindow newWindow: UIWindow?) {
+    if newWindow != nil {
+      guard let source = self.source else {
+        return
+      }
+
+      self.controller.prepare(source, view: self)
+    } else {
+      self.controller.release()
+    }
+  }
```

Comment:
As long as `removeClippedSubviews` on our `FlatList`s, only the visible items on the user's screen will ever be actively playing.

When the player leaves the viewport, we will clean up the player so as to not have a ton of players actively running. The `AVAsset` will be cached though.